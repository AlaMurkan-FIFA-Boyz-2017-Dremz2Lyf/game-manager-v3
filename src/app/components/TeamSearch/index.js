import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Grid, Header, Select, Loader, Segment, Dimmer } from 'semantic-ui-react';

const baseUrl = 'http://api.football-data.org/v1/competitions/';
const headers = new Headers({
  'X-Auth-Token': '1627141e9f1f40f4a67aae663f2e4029',
});
const options = {
  method: 'GET',
  headers,
  mode: 'cors',
};

const competitionsMap = {
  'Championship 2017/18': 'English Championship 2017/2018',
  'League One 2017/18': 'English League One 2017/2018',
  'League Two 2017/18': 'English League Two 2017/2018',
  'Primera Division 2017': 'La Liga 2017/2018',
};

class TeamSearch extends PureComponent {
  state = {
    loading: true,
    competitions: [],
    homeClubs: [],
    awayClubs: [],
    homeClub: null,
    awayClub: null,
  }

  componentWillMount() {
    fetch(`${baseUrl}?season=2017`, options).then(res => res.json()).then((competitions) => {
      this.setState({
        competitions,
        loading: false,
      });
    }).catch((error) => { throw error; });
  }

  onCompetitionChange = (_, { name, value }) => {
    const clubKey = `${name}Clubs`;
    this.setState({ loading: true });
    fetch(`${baseUrl}${value}/teams`, options).then(res => res.json()).then((clubs) => {
      this.setState({
        [clubKey]: clubs.teams,
        loading: false,
      });
    });
  }

  onClubChange = (_, target) => {
    const { name, value, text, key } = target;
    this.setState({ loading: true });

    this.setState({
      [name]: {
        value,
        text,
        key,
      },
      loading: false,
    });
  }

  mapCompName = name => (
    competitionsMap[name] ? competitionsMap[name] : name
  )

  render() {
    const { competitions, homeClubs, awayClubs, loading } = this.state;
    const competitionOptions = competitions.map(({ id, caption }) => ({
      key: id,
      value: id,
      text: this.mapCompName(caption),
    }));
    const homeClubOptions = homeClubs.map((club) => {
      const { name, crestUrl } = club;
      return {
        key: crestUrl,
        value: name,
        text: name,
      };
    });
    const awayClubOptions = awayClubs.map(({ id, name, crestUrl }) => ({
      key: crestUrl,
      value: name,
      text: name,
    }));

    return (
      <Segment basic>
        <Dimmer active={loading}>
          <Loader />
        </Dimmer>
        <Grid columns={2}>
          <Grid.Row>
            <Grid.Column width={8}>
              <Header color="grey" content="Home" />
              <Select name="home" onChange={this.onCompetitionChange} placeholder="Select a Competition" options={competitionOptions} />
              {
                homeClubOptions.length ? (
                  <Select
                    name="homeClub"
                    placeholder="Select a Club"
                    onChange={this.onClubChange}
                    options={homeClubOptions}
                  />
                ) : null
              }
            </Grid.Column>
            <Grid.Column width={8}>
              <Header color="grey" content="Away" />
              <Select name="away" onChange={this.onCompetitionChange} placeholder="Select a Competition" options={competitionOptions} />
              {
                awayClubOptions.length ? (
                  <Select
                    name="awayClub"
                    placeholder="Select a Club"
                    onChange={this.onClubChange}
                    options={awayClubOptions}
                  />
                ) : null
              }
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Segment>
    );
  }
}

TeamSearch.propTypes = {};

export default TeamSearch;
