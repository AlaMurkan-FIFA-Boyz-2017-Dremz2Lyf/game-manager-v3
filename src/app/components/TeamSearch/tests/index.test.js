import React from 'react';
import { shallow } from 'enzyme';
import fetch from 'node-fetch';

import TeamSearch from '../index';
import { competitions, premTeams } from './fixtures';

describe('<TeamSearch />', () => {
  const wrapper = shallow(<TeamSearch />);
  fetch.mockReponse(JSON.stringify({ statusCode: 200, json: () => competitions }));

  beforeEach(() => {
    fetch.reset();
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should call for the competitions in componentWillMount', () => {
    wrapper.instance().componentWillMount().then((comps) => {
      console.log({ comps });
      expect(wrapper.state('competitions')).toEqual(competitions);
      expect(wrapper.state('loading')).toBe(false);
    });
  });
});
