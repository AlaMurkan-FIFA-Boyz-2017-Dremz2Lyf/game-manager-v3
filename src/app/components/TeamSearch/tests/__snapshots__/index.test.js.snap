// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamSearch /> should match the snapshot 1`] = `
<Segment
  basic={true}
>
  <ModalContent>
    <Dimmer
      active={true}
    >
      <Loader />
    </Dimmer>
    <Grid
      columns={2}
    >
      <GridRow>
        <GridColumn
          width={8}
        >
          <Header
            color="teal"
            content="Home"
          />
          <Select
            name="home"
            onChange={[Function]}
            options={Array []}
            placeholder="Select a Competition"
          />
        </GridColumn>
        <GridColumn
          width={8}
        >
          <Header
            color="teal"
            content="Away"
          />
          <Select
            name="away"
            onChange={[Function]}
            options={Array []}
            placeholder="Select a Competition"
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </ModalContent>
  <Segment
    basic={true}
    floated="right"
  >
    <ModalActions>
      <Button
        as="button"
        basic={true}
        color="red"
        inverted={true}
      >
        <Icon
          as="i"
          name="remove"
        />
         Cancel
      </Button>
      <Button
        as="button"
        color="green"
        inverted={true}
      >
        <Icon
          as="i"
          name="checkmark"
        />
         Confirm
      </Button>
    </ModalActions>
  </Segment>
</Segment>
`;

exports[`<TeamSearch /> should match the snapshot with a home competition selected 1`] = `
<Segment
  basic={true}
>
  <ModalContent>
    <Dimmer
      active={false}
    >
      <Loader />
    </Dimmer>
    <Grid
      columns={2}
    >
      <GridRow>
        <GridColumn
          width={8}
        >
          <Header
            color="teal"
            content="Home"
          />
          <Select
            name="home"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "key": 444,
                  "text": "Campeonato Brasileiro da Série A",
                  "value": 444,
                },
                Object {
                  "key": 445,
                  "text": "Premier League 2017/18",
                  "value": 445,
                },
                Object {
                  "key": 446,
                  "text": "English Championship 2017/2018",
                  "value": 446,
                },
                Object {
                  "key": 447,
                  "text": "English League One 2017/2018",
                  "value": 447,
                },
                Object {
                  "key": 448,
                  "text": "English League Two 2017/2018",
                  "value": 448,
                },
                Object {
                  "key": 449,
                  "text": "Eredivisie 2017/18",
                  "value": 449,
                },
                Object {
                  "key": 450,
                  "text": "Ligue 1 2017/18",
                  "value": 450,
                },
                Object {
                  "key": 451,
                  "text": "Ligue 2 2017/18",
                  "value": 451,
                },
                Object {
                  "key": 452,
                  "text": "1. Bundesliga 2017/18",
                  "value": 452,
                },
                Object {
                  "key": 453,
                  "text": "2. Bundesliga 2017/18",
                  "value": 453,
                },
                Object {
                  "key": 455,
                  "text": "La Liga 2017/2018",
                  "value": 455,
                },
                Object {
                  "key": 456,
                  "text": "Serie A 2017/18",
                  "value": 456,
                },
                Object {
                  "key": 457,
                  "text": "Primeira Liga 2017/18",
                  "value": 457,
                },
                Object {
                  "key": 458,
                  "text": "DFB-Pokal 2017/18",
                  "value": 458,
                },
                Object {
                  "key": 459,
                  "text": "Serie B 2017/18",
                  "value": 459,
                },
                Object {
                  "key": 464,
                  "text": "Champions League 2017/18",
                  "value": 464,
                },
                Object {
                  "key": 466,
                  "text": "Australian A-League",
                  "value": 466,
                },
              ]
            }
            placeholder="Select a Competition"
          />
          <Select
            name="homeClub"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/en/5/53/Arsenal_FC.svg",
                  "text": "Arsenal FC",
                  "value": "Arsenal FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/en/6/63/Leicester02.png",
                  "text": "Leicester City FC",
                  "value": "Leicester City FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/e/e2/Watford.svg",
                  "text": "Watford FC",
                  "value": "Watford FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/0/0a/FC_Liverpool.svg",
                  "text": "Liverpool FC",
                  "value": "Liverpool FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/c/c9/FC_Southampton.svg",
                  "text": "Southampton FC",
                  "value": "Southampton FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/a/ab/Swansea_City_Logo.svg",
                  "text": "Swansea City FC",
                  "value": "Swansea City FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/8/8b/West_Bromwich_Albion.svg",
                  "text": "West Bromwich Albion FC",
                  "value": "West Bromwich Albion FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/de/4/41/Afc_bournemouth.svg",
                  "text": "AFC Bournemouth",
                  "value": "AFC Bournemouth",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/f/f9/Everton_FC.svg",
                  "text": "Everton FC",
                  "value": "Everton FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/a/a3/Stoke_City.svg",
                  "text": "Stoke City FC",
                  "value": "Stoke City FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/b/bf/Crystal_Palace_F.C._logo_(2013).png",
                  "text": "Crystal Palace FC",
                  "value": "Crystal Palace FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/5/5a/Huddersfield_Town_A.F.C._logo.svg",
                  "text": "Huddersfield Town",
                  "value": "Huddersfield Town",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/5/5c/Chelsea_crest.svg",
                  "text": "Chelsea FC",
                  "value": "Chelsea FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/0/02/Burnley_FC_badge.png",
                  "text": "Burnley FC",
                  "value": "Burnley FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/f/fd/Brighton_&_Hove_Albion_logo.svg",
                  "text": "Brighton & Hove Albion",
                  "value": "Brighton & Hove Albion",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/e/eb/Manchester_City_FC_badge.svg",
                  "text": "Manchester City FC",
                  "value": "Manchester City FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/5/56/Newcastle_United_Logo.svg",
                  "text": "Newcastle United FC",
                  "value": "Newcastle United FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/b/b4/Tottenham_Hotspur.svg",
                  "text": "Tottenham Hotspur FC",
                  "value": "Tottenham Hotspur FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/d/da/Manchester_United_FC.svg",
                  "text": "Manchester United FC",
                  "value": "Manchester United FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/e/e0/West_Ham_United_FC.svg",
                  "text": "West Ham United FC",
                  "value": "West Ham United FC",
                },
              ]
            }
            placeholder="Select a Club"
          />
        </GridColumn>
        <GridColumn
          width={8}
        >
          <Header
            color="teal"
            content="Away"
          />
          <Select
            name="away"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "key": 444,
                  "text": "Campeonato Brasileiro da Série A",
                  "value": 444,
                },
                Object {
                  "key": 445,
                  "text": "Premier League 2017/18",
                  "value": 445,
                },
                Object {
                  "key": 446,
                  "text": "English Championship 2017/2018",
                  "value": 446,
                },
                Object {
                  "key": 447,
                  "text": "English League One 2017/2018",
                  "value": 447,
                },
                Object {
                  "key": 448,
                  "text": "English League Two 2017/2018",
                  "value": 448,
                },
                Object {
                  "key": 449,
                  "text": "Eredivisie 2017/18",
                  "value": 449,
                },
                Object {
                  "key": 450,
                  "text": "Ligue 1 2017/18",
                  "value": 450,
                },
                Object {
                  "key": 451,
                  "text": "Ligue 2 2017/18",
                  "value": 451,
                },
                Object {
                  "key": 452,
                  "text": "1. Bundesliga 2017/18",
                  "value": 452,
                },
                Object {
                  "key": 453,
                  "text": "2. Bundesliga 2017/18",
                  "value": 453,
                },
                Object {
                  "key": 455,
                  "text": "La Liga 2017/2018",
                  "value": 455,
                },
                Object {
                  "key": 456,
                  "text": "Serie A 2017/18",
                  "value": 456,
                },
                Object {
                  "key": 457,
                  "text": "Primeira Liga 2017/18",
                  "value": 457,
                },
                Object {
                  "key": 458,
                  "text": "DFB-Pokal 2017/18",
                  "value": 458,
                },
                Object {
                  "key": 459,
                  "text": "Serie B 2017/18",
                  "value": 459,
                },
                Object {
                  "key": 464,
                  "text": "Champions League 2017/18",
                  "value": 464,
                },
                Object {
                  "key": 466,
                  "text": "Australian A-League",
                  "value": 466,
                },
              ]
            }
            placeholder="Select a Competition"
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </ModalContent>
  <Segment
    basic={true}
    floated="right"
  >
    <ModalActions>
      <Button
        as="button"
        basic={true}
        color="red"
        inverted={true}
      >
        <Icon
          as="i"
          name="remove"
        />
         Cancel
      </Button>
      <Button
        as="button"
        color="green"
        inverted={true}
      >
        <Icon
          as="i"
          name="checkmark"
        />
         Confirm
      </Button>
    </ModalActions>
  </Segment>
</Segment>
`;

exports[`<TeamSearch /> should match the snapshot with a home competition selected 2`] = `
<Segment
  basic={true}
>
  <ModalContent>
    <Dimmer
      active={false}
    >
      <Loader />
    </Dimmer>
    <Grid
      columns={2}
    >
      <GridRow>
        <GridColumn
          width={8}
        >
          <Header
            color="teal"
            content="Home"
          />
          <Select
            name="home"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "key": 444,
                  "text": "Campeonato Brasileiro da Série A",
                  "value": 444,
                },
                Object {
                  "key": 445,
                  "text": "Premier League 2017/18",
                  "value": 445,
                },
                Object {
                  "key": 446,
                  "text": "English Championship 2017/2018",
                  "value": 446,
                },
                Object {
                  "key": 447,
                  "text": "English League One 2017/2018",
                  "value": 447,
                },
                Object {
                  "key": 448,
                  "text": "English League Two 2017/2018",
                  "value": 448,
                },
                Object {
                  "key": 449,
                  "text": "Eredivisie 2017/18",
                  "value": 449,
                },
                Object {
                  "key": 450,
                  "text": "Ligue 1 2017/18",
                  "value": 450,
                },
                Object {
                  "key": 451,
                  "text": "Ligue 2 2017/18",
                  "value": 451,
                },
                Object {
                  "key": 452,
                  "text": "1. Bundesliga 2017/18",
                  "value": 452,
                },
                Object {
                  "key": 453,
                  "text": "2. Bundesliga 2017/18",
                  "value": 453,
                },
                Object {
                  "key": 455,
                  "text": "La Liga 2017/2018",
                  "value": 455,
                },
                Object {
                  "key": 456,
                  "text": "Serie A 2017/18",
                  "value": 456,
                },
                Object {
                  "key": 457,
                  "text": "Primeira Liga 2017/18",
                  "value": 457,
                },
                Object {
                  "key": 458,
                  "text": "DFB-Pokal 2017/18",
                  "value": 458,
                },
                Object {
                  "key": 459,
                  "text": "Serie B 2017/18",
                  "value": 459,
                },
                Object {
                  "key": 464,
                  "text": "Champions League 2017/18",
                  "value": 464,
                },
                Object {
                  "key": 466,
                  "text": "Australian A-League",
                  "value": 466,
                },
              ]
            }
            placeholder="Select a Competition"
          />
        </GridColumn>
        <GridColumn
          width={8}
        >
          <Header
            color="teal"
            content="Away"
          />
          <Select
            name="away"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "key": 444,
                  "text": "Campeonato Brasileiro da Série A",
                  "value": 444,
                },
                Object {
                  "key": 445,
                  "text": "Premier League 2017/18",
                  "value": 445,
                },
                Object {
                  "key": 446,
                  "text": "English Championship 2017/2018",
                  "value": 446,
                },
                Object {
                  "key": 447,
                  "text": "English League One 2017/2018",
                  "value": 447,
                },
                Object {
                  "key": 448,
                  "text": "English League Two 2017/2018",
                  "value": 448,
                },
                Object {
                  "key": 449,
                  "text": "Eredivisie 2017/18",
                  "value": 449,
                },
                Object {
                  "key": 450,
                  "text": "Ligue 1 2017/18",
                  "value": 450,
                },
                Object {
                  "key": 451,
                  "text": "Ligue 2 2017/18",
                  "value": 451,
                },
                Object {
                  "key": 452,
                  "text": "1. Bundesliga 2017/18",
                  "value": 452,
                },
                Object {
                  "key": 453,
                  "text": "2. Bundesliga 2017/18",
                  "value": 453,
                },
                Object {
                  "key": 455,
                  "text": "La Liga 2017/2018",
                  "value": 455,
                },
                Object {
                  "key": 456,
                  "text": "Serie A 2017/18",
                  "value": 456,
                },
                Object {
                  "key": 457,
                  "text": "Primeira Liga 2017/18",
                  "value": 457,
                },
                Object {
                  "key": 458,
                  "text": "DFB-Pokal 2017/18",
                  "value": 458,
                },
                Object {
                  "key": 459,
                  "text": "Serie B 2017/18",
                  "value": 459,
                },
                Object {
                  "key": 464,
                  "text": "Champions League 2017/18",
                  "value": 464,
                },
                Object {
                  "key": 466,
                  "text": "Australian A-League",
                  "value": 466,
                },
              ]
            }
            placeholder="Select a Competition"
          />
          <Select
            name="awayClub"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/en/5/53/Arsenal_FC.svg",
                  "text": "Arsenal FC",
                  "value": "Arsenal FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/en/6/63/Leicester02.png",
                  "text": "Leicester City FC",
                  "value": "Leicester City FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/e/e2/Watford.svg",
                  "text": "Watford FC",
                  "value": "Watford FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/0/0a/FC_Liverpool.svg",
                  "text": "Liverpool FC",
                  "value": "Liverpool FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/c/c9/FC_Southampton.svg",
                  "text": "Southampton FC",
                  "value": "Southampton FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/a/ab/Swansea_City_Logo.svg",
                  "text": "Swansea City FC",
                  "value": "Swansea City FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/8/8b/West_Bromwich_Albion.svg",
                  "text": "West Bromwich Albion FC",
                  "value": "West Bromwich Albion FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/de/4/41/Afc_bournemouth.svg",
                  "text": "AFC Bournemouth",
                  "value": "AFC Bournemouth",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/f/f9/Everton_FC.svg",
                  "text": "Everton FC",
                  "value": "Everton FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/a/a3/Stoke_City.svg",
                  "text": "Stoke City FC",
                  "value": "Stoke City FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/b/bf/Crystal_Palace_F.C._logo_(2013).png",
                  "text": "Crystal Palace FC",
                  "value": "Crystal Palace FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/5/5a/Huddersfield_Town_A.F.C._logo.svg",
                  "text": "Huddersfield Town",
                  "value": "Huddersfield Town",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/5/5c/Chelsea_crest.svg",
                  "text": "Chelsea FC",
                  "value": "Chelsea FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/0/02/Burnley_FC_badge.png",
                  "text": "Burnley FC",
                  "value": "Burnley FC",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/f/fd/Brighton_&_Hove_Albion_logo.svg",
                  "text": "Brighton & Hove Albion",
                  "value": "Brighton & Hove Albion",
                },
                Object {
                  "key": "https://upload.wikimedia.org/wikipedia/en/e/eb/Manchester_City_FC_badge.svg",
                  "text": "Manchester City FC",
                  "value": "Manchester City FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/5/56/Newcastle_United_Logo.svg",
                  "text": "Newcastle United FC",
                  "value": "Newcastle United FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/b/b4/Tottenham_Hotspur.svg",
                  "text": "Tottenham Hotspur FC",
                  "value": "Tottenham Hotspur FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/d/da/Manchester_United_FC.svg",
                  "text": "Manchester United FC",
                  "value": "Manchester United FC",
                },
                Object {
                  "key": "http://upload.wikimedia.org/wikipedia/de/e/e0/West_Ham_United_FC.svg",
                  "text": "West Ham United FC",
                  "value": "West Ham United FC",
                },
              ]
            }
            placeholder="Select a Club"
          />
        </GridColumn>
      </GridRow>
    </Grid>
  </ModalContent>
  <Segment
    basic={true}
    floated="right"
  >
    <ModalActions>
      <Button
        as="button"
        basic={true}
        color="red"
        inverted={true}
      >
        <Icon
          as="i"
          name="remove"
        />
         Cancel
      </Button>
      <Button
        as="button"
        color="green"
        inverted={true}
      >
        <Icon
          as="i"
          name="checkmark"
        />
         Confirm
      </Button>
    </ModalActions>
  </Segment>
</Segment>
`;
